include/kernel/stats.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2021, Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_KERNEL_STATS_H_
+#define ZEPHYR_INCLUDE_KERNEL_STATS_H_
+
+#include <stdint.h>
+
+/*
+ * [k_cycle_stats] is used to track internal statistics about both thread
+ * and CPU usage.
+ */
+
+struct k_cycle_stats {
```

Comment:
This seems like it could use the stats subsys here. It would allow exporting these with mcumgr initially and potentially more exporters in the future.

Reply:
I am not familiar with that subsys yet.

Reply:
I agree with @teburd. The stats subsystem can also be transferred over mcumgr after, which is very useful.

Reply:
I've been reading up on the stats subsys a bit, and I think this does bear looking into more. It could be a handy enhancement for a subsequent work item.